src/libmw/libmw_node.cpp

```diff
@@ -98,6 +99,25 @@ MWEXPORT void CheckTransaction(const libmw::TxRef& transaction)
     BlockSumValidator::ValidateForTx(*transaction.pTransaction);
 }
 
+MWEXPORT void CheckTxInputs(const libmw::TxRef& transaction, const libmw::CoinsViewRef& view, int nSpendHeight)
+{
+    assert(transaction.pTransaction != nullptr);
+    assert(view.pCoinsView != nullptr);
+
+    auto pCoinsView = std::dynamic_pointer_cast<mw::CoinsViewCache>(view.pCoinsView);
+    assert(pCoinsView != nullptr);
+
+    for (const Input& input : transaction.pTransaction->GetInputs()) {
+        auto utxos = pCoinsView->GetUTXOs(input.GetCommitment());
+        if (utxos.empty()) {
+            ThrowValidation(EConsensusError::UTXO_MISSING);
+        }
+        if (input.IsPeggedIn() && nSpendHeight - utxos.back()->GetBlockHeight() < mw::ChainParams::GetPegInMaturity()) {
```

Comment:
I'm surprised this doesn't give warnings about `nSpendHeight - utxos.back()->GetBlockHeight()`. GetBlockHeight() returns a u64.

I have 2 suggestions:

1. Change nSpendHeight to be a uint64_t, for consistency with all of our other APIs.
2. Check if `nSpendHeight < (utxos.back()->GetBlockHeight() + mw::ChainParams::GetPegInMaturity())` instead, so we don't have to worry about the rare case where GetBlockHeight() is actually higher than nSpendHeight.